tests/test_utils.py

```diff
@@ -558,9 +579,16 @@ def initialize_git_repo(rpath, files=[]):
     subprocess.Popen(['git', 'init', rpath])
     subprocess.Popen(['git', 'config', 'user.name', '"Gerald Host"'], cwd=rpath)
     subprocess.Popen(['git', 'config', 'user.email', '"ghost@example.com"'], cwd=rpath)
+    first_commit_ref = None
     for f in files:
+        subprocess.Popen(['touch', f], cwd=rpath)
         subprocess.Popen(['git', 'add', f], cwd=rpath)
+        subprocess.Popen(['git', 'commit', '-m', 'new file {0}'.format(f)], cwd=rpath)
+        if not first_commit_ref:
+            shortlog = subprocess.check_output(['git', 'log', '--oneline'], cwd=rpath)
+            first_commit_ref = shortlog.split(" ")[0]
```

Comment:
This fails for python 3 [1] with `TypeError: a bytes-like object is required, not 'str'` since check_output returns a byte string. You could check the `universal_newlines` param [2] or perform a conversion (using a byte literal b' ' should be ignored in python 2.

[1] https://travis-ci.org/projectatomic/osbs-client/jobs/485979150
[2] https://docs.python.org/3/library/subprocess.html#subprocess.check_output

Reply:
using rev-parse instead.